In [1]:
import os
if 'jbook' in os.getcwd():
    os.chdir(os.path.abspath(os.path.join("../..")))
import warnings
warnings.filterwarnings("ignore")

In [2]:
# from importlib import reload  # Not needed in Python 2
# import logging
# reload(logging)
# logging.basicConfig(format='%(asctime)s %(levelname)s:%(message)s', level=logging.DEBUG, datefmt='%I:%M:%S')

# Normalization

To prepare the dataset for analysis, normalization preprocessing is performed. This involves standardizing data types, ensuring consistent encoding, and removing new lines from text. These steps help resolve technical anomalies and streamline subsequent data processing and analysis.

In [3]:
from appinsight.data_prep.normalize import NormalizeConfig, Normalize

To implement the normalization preprocessing described above, the following code snippet is utilized. It initializes the normalization configuration, executes the normalization process, and then displays information about the resulting dataset using the `info()` function.

In [5]:
config = NormalizeConfig(force=False)
norm = Normalize(config=config)
data = norm.execute()
data.info(show_counts=True, memory_usage=True, verbose=True)

<class 'pandas.core.frame.DataFrame'>
Index: 22166591 entries, 8082666 to 21984044
Data columns (total 11 columns):
 #   Column       Non-Null Count     Dtype         
---  ------       --------------     -----         
 0   id           22166591 non-null  string        
 1   app_id       22166591 non-null  string        
 2   app_name     22166591 non-null  string        
 3   category_id  22166591 non-null  category      
 4   category     22166591 non-null  category      
 5   author       22166591 non-null  string        
 6   rating       22166591 non-null  int16         
 7   content      22166591 non-null  string        
 8   vote_sum     22166591 non-null  int64         
 9   vote_count   22166591 non-null  int64         
 10  date         22166591 non-null  datetime64[ms]
dtypes: category(2), datetime64[ms](1), int16(1), int64(2), string(5)
memory usage: 1.6 GB


Datatypes have been successfully cast, utf-8 encoding verified, and newlines have been removed, yielding a dataset of 22,166,591 observations and no null values.